packages/manager/modules/sms/src/sms/options/smppParameter/smppParameter.controller.js

```diff
@@ -1,19 +1,44 @@
 export default class SmsSmppParameterCtrl {
   /* @ngInject */
-  constructor(SmsService) {
+  constructor(SmsService, $timeout, $state) {
     this.smsService = SmsService;
+    this.$timeout = $timeout;
+    this.$state = $state;
   }
 
   $onInit() {
+    this.checkStatus = this.checkStatus.bind(this);
     this.OPTION_SMPP_TRACKING_PREFIX = 'sms::service::configure-smpp::';
     this.loading = true;
     return this.smsService
       .getSmppSettings(this.serviceName)
       .then((result) => {
         this.smppSettings = result;
+        if (this.smppSettings.status === 'UPDATING_IP') {
+          this.checkStatus();
+        }
       })
       .finally(() => {
         this.loading = false;
       });
   }
+
+  $onDestroy() {
+    this.checkStatus = () => false;
```

Comment:
**question:** Why redefining the function in `onDestroy`?  Doesn't smell good.

Reply:
Without this, if the user goes to the page with a service in UPDATE_IP, it starts the function (which is desired) but if he goes out and comes back several times in a few seconds, the function is started several times, whereas in this case the polling function is killed if he goes out of the page.

Reply:
I did modify $onDestroy